app/src/main/java/com/example/chessclockk/MainActivity.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.chessclockk
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.example.chessclockk.screens.main.MainScreen
+import com.example.chessclockk.ui.theme.ChessClockkTheme
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class MainActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            ChessClockkTheme {
+                Surface(
```

Comment:
Najlepiej wynosić jak najwięcej Composable poza klasę. Tam pod spodem jest ukryta referencja na "this". A to wpływa na rekompozycję. Fragmentowe "this" będzie od razu traktowane jako argument typu "stable" https://developer.android.com/develop/ui/compose/performance/stability#types